platform_channels/lib/src/pet_list_screen.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:platform_channels/src/pet_list_message_channel.dart';
+
+/// Demonstrates how to use [BasicMessageChannel] to send & receive the platform
+/// Message.
+class PetListScreen extends StatefulWidget {
+  @override
+  _PetListScreenState createState() => _PetListScreenState();
+}
+
+class _PetListScreenState extends State<PetListScreen> {
+  PetModel petModel;
+
+  @override
+  void initState() {
+    super.initState();
+    // Receives a string of json object from the platform and converts it
+    // to PetModel.
+    BasicMessageChannel('stringCodecDemo', StringCodec())
+        .setMessageHandler((message) async {
+      setState(() {
+        petModel = PetModel.fromJson(message);
+      });
+      return;
+    });
```

Comment:
Hey, @RedBrogdon I'm not sure how can I test this part with the UI. 

I had an idea to break these tests into two parts. One unit test with a Mock to see, the JSON is converted to `PetModel` and `ListView` is properly rendered when a `PetMode.petList` is given.

Reply:
See my response below. I think splitting things up is definitely a good idea.